services/profile/profile.go

```diff
@@ -49,3 +49,33 @@ func (ctx Context) GetCompanies(user string) ([]types.Company, error) {
 
 	return ret.Details, nil
 }
+
+// GetOptions retrieves all options for the specified user
+func (ctx Context) GetOptions(user string) (map[string]interface{}, error) {
+
+	optionsURL := fmt.Sprintf("%s/options", ctx.URL)
+
+	req, err := http.Request{
+		URL:     optionsURL,
+		Authentication: http.Authentication{
+			Scheme:   "basic",
+			Username: ctx.ClientID,
+			Password: ctx.ClientSecret,
+		},
+		Headers: map[string]string{
+			"x-9sp-user": user,
+		},
+		ContentType: "application/json",
+	}.Get()
+
+	if err != nil {
+		return nil, fmt.Errorf("Error getting user profile data %s", err.Error())
+	}
+
+	var profileOptions map[string]interface{}
```

Comment:
This should be unmarshaled into a `api.Response` struct.  We want to evaluate the value of `status` to be `ok` before responding.  The response should only include the `.details` interface

Reply:
Using `api.Response` for `GetOptions` and `GetOption`. I didn't make the change in `GetCompanies` because it seems to be more complicated to cast an array. I tried doing  `ret.Details.([]types.Company)`

but I got this error
```
panic: interface conversion: interface {} is []interface {}, not []types.Company [recovered]
        panic: interface conversion: interface {} is []interface {}, not []types.Company
```

I seems that you need to convert first to a interface{} array and then to a Company.